File: src/main/kotlin/space/davids_digital/kiri/rest/dto/telegram/TelegramBroadcastDto.kt
package space.davids_digital.kiri.rest.dto.telegram

import java.time.ZonedDateTime
import java.util.UUID

data class TelegramBroadcastDto(
    val id: UUID,
    val status: Status,
    val silent: Boolean,
    val total: Int,
    val sent: Int,
    val failed: Int,
    val failures: List<Failure>,
    val startedAt: ZonedDateTime,
    val finishedAt: ZonedDateTime?,
    val error: String?,
) {
    data class Failure(
        val chatId: Long,
        val chatTitle: String?,
        val error: String,
    )

    enum class Status {
        RUNNING, COMPLETED, FAILED
    }
}

File: admin-ui/lib/api/types/telegram/TelegramBroadcastDto.ts
export type TelegramBroadcastDto = {
    id: string;
    status: TelegramBroadcastDto.Status;
    silent: boolean;
    total: number;
    sent: number;
    failed: number;
    failures: TelegramBroadcastDto.Failure[];
    startedAt: string;
    finishedAt: string | null;
    error: string | null;
};

export namespace TelegramBroadcastDto {
    export type Status = "RUNNING" | "COMPLETED" | "FAILED";

    export type Failure = {
        chatId: number;
        chatTitle: string | null;
        error: string;
    };
}

export type TelegramBroadcastStatusDto = {
    recipients: number;
    latest: TelegramBroadcastDto | null;
};

export type TelegramBroadcastRequest = {
    text: string;
    silent: boolean;
};
